File: src/LavaZone.java
import processing.core.PApplet;

/** A rectangular hazard zone (lava/ice-hole/acid depending on the map) that damages balls standing in it. */
public class LavaZone {
    float x, y, w, h;

    public LavaZone(float x, float y, float w, float h) {
        this.x = x;
        this.y = y;
        this.w = w;
        this.h = h;
    }

    public void checkDamage(Ball b) {
        if (b.x > x - w / 2 && b.x < x + w / 2 && b.y > y - h / 2 && b.y < y + h / 2) {
            b.takeDamage(0.3f);
        }
    }

    /** fillColor/strokeColor come from the current map's palette (GameConfig.HAZARD_FILL / HAZARD_LINE). */
    public void draw(PApplet p, int fillColor, int strokeColor) {
        p.pushMatrix();
        p.translate(x, y, -1);
        p.rectMode(PApplet.CENTER);
        p.fill(fillColor);
        p.stroke(strokeColor);
        p.strokeWeight(3);
        p.rect(0, 0, w, h, 15);
        p.popMatrix();
    }
}

File: LavaZone.js
/* global p5, Ball */

class LavaZone {
    constructor(x, y, w, h) {
        this.x = x;
        this.y = y;
        this.w = w;
        this.h = h;
    }

    checkDamage(b) {
        if (b.x > this.x - this.w / 2 && b.x < this.x + this.w / 2 &&
            b.y > this.y - this.h / 2 && b.y < this.y + this.h / 2) {
            b.takeDamage(0.3);
        }
    }

    /** fillColor/strokeColor come from the current map's palette (GameConfig.HAZARD_FILL / HAZARD_LINE). */
    draw(fillColor, strokeColor) {
        push();
        translate(this.x, this.y, -1);
        rectMode(CENTER);
        fill(fillColor);
        stroke(strokeColor);
        strokeWeight(3);
        rect(0, 0, this.w, this.h, 15);
        pop();
    }
}